apps/cowswap-frontend/src/modules/swap/services/ethFlow/index.ts

```diff
@@ -78,6 +82,10 @@ export async function ethFlow(
       throw new Error('Quote expired. Please refresh.')
     }
 
+    if (contract.address !== getEthFlowContractAddress(chainId)) {
+      throw new Error('EthFlow contract address mismatch. Please refresh the page and try again.')
+    }
```

Comment:
This is the main change

If somehow the contract address doesn't match the one that should be used for the chain, prevent user from continuing.

Reply:
What if chainId is wrong here?
I mean, user is on Mainnet but for some reason contract comes from Arbitrum and `chainId` here is also Arbitrum

Reply:
Maybe you can make sure inside the method `ethFlowContract.createOrder`, you can make a check of what is the `chainId` of `ethFlowContract.provider`?

Reply:
> What if chainId is wrong here?
Well, good point.

I'll try Anxo's idea to solve this.

Reply:
@anxolin @shoom3301 added another conditional which checks for the current chain from the wallet https://github.com/cowprotocol/cowswap/pull/4629/commits/5c794167f544bb950d5c4db0e956836ec222fd24